validation-test/BuildSystem/swift-cmake/CMakeLists.txt

```diff
@@ -23,13 +23,27 @@ add_swift_host_library(TestPureSwiftSharedLib
   PURE_SWIFT
   Klass.swift)
 target_link_libraries(TestPureSwiftSharedLib PRIVATE TestCPPLib)
+# FIXME: cross-compile for all variants.
+target_link_libraries(TestPureSwiftSharedLib
+    PRIVATE
+    swiftCore${SWIFT_PRIMARY_VARIANT_SUFFIX}
```

Comment:
I think this is incorrect. This should happen automagically from the host side swift. If there are issues on your platform, I would rather just block this on being Darwin for a little bit and we should talk.

Reply:
Actually let me be more specific. I think the problem is this:

```
      if (CMAKE_Swift_COMPILER)                                                                                               
        # Add in the SDK directory for the host platform and add an rpath.                                                    
        target_link_directories(${name} PRIVATE                                                                               
          ${SWIFT_SDK_${SWIFT_HOST_VARIANT_SDK}_ARCH_${SWIFT_HOST_VARIANT_ARCH}_PATH}/usr/lib/swift)                          
        # Add in the toolchain directory so we can grab compatibility libraries                                               
        get_filename_component(TOOLCHAIN_BIN_DIR ${CMAKE_Swift_COMPILER} DIRECTORY)                                           
        get_filename_component(TOOLCHAIN_LIB_DIR "${TOOLCHAIN_BIN_DIR}/../lib/swift/macosx" ABSOLUTE)                         
        target_link_directories(${name} PUBLIC ${TOOLCHAIN_LIB_DIR})                                                          
      endif() 
```

I added this line so that on macOS we could find the relevant libraries. I just realized that since we do not have a toolchain on the bots for Linux that we didn't actually test this on Linux! This whole test/code path needs to be conditionalized on being Darwin (I had not thought about that).